src/commands/Suggestions/resolvesuggestion.ts

```diff
@@ -0,0 +1,122 @@
+import { CommandStore, KlasaMessage } from 'klasa';
+import { MessageEmbed, TextChannel } from 'discord.js';
+import { SkyraCommand } from '@lib/structures/SkyraCommand';
+import { PermissionLevels } from '@lib/types/Enums';
+import { GuildSettings } from '@lib/types/settings/GuildSettings';
+import { isNumber } from '@klasa/utils';
+import { SuggestionData } from '@lib/types/definitions/Suggestion';
+
+const enum SuggestionsColors {
+	Accepted = '4CB02C',
+	Considered = 'CFA08D',
+	Denied = 'F90505'
+}
+
+export default class extends SkyraCommand {
+
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			aliases: ['resu', 'rs'],
+			cooldown: 10,
+			description: language => language.tget('COMMAND_RESOLVESUGGESTION_DESCRIPTION'),
+			extendedHelp: language => language.tget('COMMAND_RESOLVESUGGESTION_EXTENDED'),
+			flagSupport: true,
+			permissionLevel: PermissionLevels.Moderator,
+			requiredPermissions: ['EMBED_LINKS'],
+			runIn: ['text'],
+			usage: '<suggestion:suggestion> <accept|a|deny|d|consider|c> [comment:...string]',
+			usageDelim: ' '
+		});
+
+		this.createCustomResolver('suggestion', async (arg, _, message) => {
+			const channelID = message.guild!.settings!.get(GuildSettings.Suggestions.SuggestionsChannel);
+			if (channelID === null) throw message.language.tget('COMMAND_SUGGEST_NOSETUP', message.author.username);
+			const channel = this.client.channels.get(channelID) as TextChannel;
+
+			const id = parseInt(arg, 10);
+			if (!isNumber(id)) throw `That's not a valid suggestion ID!`;
+
+			const suggestionData = await this.client.queries.fetchSuggestion(message.guild!.id, id);
+			if (suggestionData === null) throw `Couldn't find a suggestion with that ID`;
+
+			const suggestionMessage = channel.messages.get(suggestionData.message_id);
+			console.log(suggestionMessage);
+			if (typeof suggestionMessage === 'undefined') {
+				await this.client.queries.deleteSuggestion(message.guild!.id, id);
+				throw `Couldn't find the suggestion's message. To prevent clutter, I have removed it from the database.`;
+			}
+
+
+			const suggestionAuthor = await this.client.users.fetch(suggestionData.author_id).catch(() => null);
+
+			return {
+				message: suggestionMessage,
+				author: suggestionAuthor,
+				id
+			} as SuggestionData;
+		});
+	}
+
+	public async run(message: KlasaMessage, [suggestionData, action, comment]: [SuggestionData, string, string | undefined]) {
+		const [shouldDM, shouldHideAuthor, shouldRepostSuggestion] = message.guild!.settings.pluck(GuildSettings.Suggestions.OnAction.DM, GuildSettings.Suggestions.OnAction.HideAuthor, GuildSettings.Suggestions.OnAction.RepostMessage) as boolean[];
```

Comment:
You know, I'm not fond of using pluck here for the mere fact it's weakly typed.

Reply:
huh? I mean, every value is a boolean, and it's typed as such